File: patterns/java/src/refactoring_guru/patterns/abstract_factory/examples/Application.java
package refactoring_guru.patterns.abstract_factory.examples;


import refactoring_guru.patterns.abstract_factory.examples.button.Button;
import refactoring_guru.patterns.abstract_factory.examples.factory.GUIFactory;

public class Application {
    private Button button;

    public Application(GUIFactory factory) {
        this.button = factory.createButton();
    }

    public Button getButton() {
        return button;
    }
}

File: patterns/php/src/AbstractFactory/Example/Application.php
<?php

namespace RefactoringGuru\Patterns\AbstractFactory\Example;

use RefactoringGuru\Patterns\AbstractFactory\Example\Factories\GUIFactoryInterface;

class Application
{
    private $button;

    private $checkbox;

    public function __construct(GUIFactoryInterface $factory)
    {
        $this->button = $factory->createButton();
        $this->checkbox = $factory->createCheckbox();
    }

    public function paint()
    {
        $this->button->paint();
        $this->checkbox->paint();
    }
}
